### PR TITLE
Request reviewers automatically

### LINKED ISSUE #2: Set backport PR reviewer equal to original PR reviewer
Original issue: https://github.com/zeebe-io/automation-lab/issues/39

### PR DESCRIPTION
**Pull request by korthout**
This PR makes the action request reviewers for the newly created backport PRs. For this it uses the same reviewers as the PR that is backported.

closes #2

### CODE REVIEW

**Comment by korthout** [comment]
Test in automation lab successful 

source PR: https://github.com/zeebe-io/automation-lab/pull/47
created backport PR: https://github.com/zeebe-io/automation-lab/pull/48
request for reviewers: https://github.com/zeebe-io/automation-lab/pull/48#event-4046885226

**Comment by korthout** [comment]
@pihme Please have a look

File: src/main.ts [hunk 1]
```diff
@@ -2,8 +2,13 @@ import * as core from "@actions/core";
...
 import * as github from "@actions/github";
 import { EventPayloads } from "@octokit/webhooks";
-import { OctokitResponse, PullsCreateResponseData } from "@octokit/types";
+import {
+  OctokitResponse,
+  PullsCreateResponseData,
+  PullsRequestReviewersResponseData,
+} from "@octokit/types";
 import dedent from "dedent";
+import { resolve } from "path";
```
Comment (pihme) [inline src/main.ts:11]: I cannot find where this is used. Am I missing something?

File: src/main.ts [hunk 2]
```diff
@@ -40,11 +52,11 @@ async function run(): Promise<void> {
...
-    const issue_number = payload.pull_request.number;
-    const issue_title = payload.pull_request.title;
-    const headref = payload.pull_request.head.sha;
-    const baseref = payload.pull_request.base.sha;
-    const labels = payload.pull_request.labels;
+    const pr = payload.pull_request;
+    const headref = pr.head.sha;
+    const baseref = pr.base.sha;
+    const labels = pr.labels;
+    const reviewers = pr.requested_reviewers.map((r) => r.login);
```
Comment (pihme) [inline src/main.ts:59]: Not sure if this is null-safe. What happens if a PR has no reviewers?

File: src/main.ts [hunk 3]
```diff
@@ -40,11 +52,11 @@ async function run(): Promise<void> {
     const owner = github.context.repo.owner;
     const repo = payload.repository.name;
 
-    const issue_number = payload.pull_request.number;
-    const issue_title = payload.pull_request.title;
-    const headref = payload.pull_request.head.sha;
-    const baseref = payload.pull_request.base.sha;
-    const labels = payload.pull_request.labels;
+    const pr = payload.pull_request;
```
Comment (pihme) [inline src/main.ts:55]: Maybe rename this to `original_pr` or something like that. Further below it gets confusing which PR is meant

**Review by pihme — APPROVED** [review]
Looks great. Some minor points to consider before merging. None of those would be a bug, so feel free to implement or ignore them.

**Comment by pihme** [comment]
I think we should also start implementing tests going forward.

File: src/main.ts (hunk 2 above)
Comment (korthout) [reply src/main.ts]: Good point. I've applied the [optional-chaining operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) in combination with the [nullish-coalescing operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) to make this null-safe.

File: src/main.ts (hunk 3 above)
Comment (korthout) [reply src/main.ts]: I like the idea, but went for `mainpr` which is a bit shorter and causes less line wrappings.

**Comment by korthout** [comment]
@pihme I agree we should add tests. Should we do that before merging this or do you think it's out of scope? After applying the review hints I now still need to do an additional manual test to see that it still works. 😞

**Comment by pihme** [comment]
@korthout Let's add the tests for the next change.

**Pull request merged**